kernel/src/loader/mod.rs

```diff
@@ -216,60 +213,16 @@ where
         .map_err(|_| Error::SeekKernelImage)?;
 
     kernel_load_offset = kernel_load_offset + start_address;
-    guest_mem
-        .read_to_memory(
-            GuestAddress(kernel_load_offset),
-            kernel_image,
-            kernel_size as usize,
-        )
-        .map_err(|_| Error::ReadKernelImage)?;
+    read_to_memory(kernel_load_offset, kernel_image, kernel_size as usize)?;
 
-    Ok(GuestAddress(kernel_load_offset))
-}
-
-/// Writes the command line string to the given memory slice.
-///
-/// # Arguments
-///
-/// * `guest_mem` - A u8 slice that will be partially overwritten by the command line.
-/// * `guest_addr` - The address in `guest_mem` at which to load the command line.
-/// * `cmdline` - The kernel command line.
-pub fn load_cmdline(
```

Comment:
I am not sure if moving loading the command line to the vmm is the right choice. To me it looks like moving the refactoring problem elsewhere.
I assume this move was necessary to get rid of the memory model dependency, but I am wondering if we can create a better abstraction here. I don't have anything in mind at the moment, but we can take this offline and see if we can come up with a better approach.

Reply:
Actually I've put in quite a lot of thought in this part. I think the vmm is the right place for these particular lines of code.

This is simply _vmm-glue_ code that just copies the contents of a `cmdline` which is an external vmm-agnostic implementation of a "kernel cmdline" into `guest_memory` which is also implemented vmm-agnostic in a different crate.

Kernel-cmdline logic should not exist in the memory_model, and memory_model logic is not necessary in the kernel cmdline logic.

Vmm crate looks like the right place for the gluing function.

I was also conflicted on whether `vmm::load_kernel_cmdline()` (the glue function) should be a static one or should take `self`. I've decided going with the latter and used it accordingly.